updater/src/updater.rs

```diff
@@ -246,17 +297,6 @@ impl Updater {
 			return;
 		}
 
-		if self.operations.lock().is_none() {
-			if let Some(ops_addr) = self.client.upgrade().and_then(|c| c.registry_address("operations".into())) {
-				trace!(target: "updater", "Found operations at {}", ops_addr);
-				let client = self.client.clone();
-				*self.operations.lock() = Some(Operations::new(ops_addr, move |a, d| client.upgrade().ok_or("No client!".into()).and_then(|c| c.call_contract(BlockId::Latest, a, d))));
```

Comment:
I haven't investigated that, but this lazy evaluation could lead to another retain cycle.

`updater` owns `operations`. `operations` owns `client`

but it was not clearly visible, since `updater` owns a weak reference to a `client`